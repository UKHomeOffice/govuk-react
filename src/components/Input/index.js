// https://github.com/alphagov/govuk-frontend/tree/master/src/components/
import React from "react";
import glamorous from "glamorous";
import PropTypes from "prop-types";

import { BLACK, YELLOW, ERROR_COLOUR } from "govuk-colours";
import { FONT_SIZE, MEDIA_QUERIES, NTA_LIGHT } from "../../constants/index";

const InputText = glamorous.input(
  {
    boxSizing: "border-box",
    fontFamily: NTA_LIGHT,
    WebkitFontSmoothing: "antialiased",
    MozOsxFontSmoothing: "grayscale",
    fontWeight: 400,
    textTransform: "none",
    fontSize: FONT_SIZE.SIZE_16,
    lineHeight: "20px",
    width: "100%",
    padding: "5px 4px 4px",
    border: `2px solid ${BLACK}`,
    [MEDIA_QUERIES.LARGESCREEN]: {
      width: "50%",
      fontSize: FONT_SIZE.SIZE_19,
      lineHeight: "25px"
    },
    "[disabled]": {
      cursor: "auto"
    },
    ":focus": {
      outline: `3px solid ${YELLOW}`,
      outlineOffset: 0
    }
  },
  ({ errorText }) => ({
    border: errorText ? `4px solid ${ERROR_COLOUR}` : `2px solid ${BLACK}`
  })
);

const Input = ({ value, onChange }) => (
  <InputText type="text" value={value} onChange={onChange} />
);

Input.defaultProps = {
  value: undefined,
  onChange: null
};

Input.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func
};

export default Input;
