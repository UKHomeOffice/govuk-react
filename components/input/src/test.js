import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { ORANGE } from 'govuk-colours';
import Input from './';

describe('Input', () => {
  const wrapper = <Input type="text" />;
  const withError = <Input type="text" error="example" />;
  const withColouredError = <Input type="text" errorColour={ORANGE} error="example" />;

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('renders with error', () => {
    const div = document.createElement('div');
    ReactDOM.render(withError, div);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });

  it('matches withError snapshot', () => {
    expect(mount(withError)).toMatchSnapshot('with error mount');
  });

  it('matches withColouredError snapshot', () => {
    expect(mount(withColouredError)).toMatchSnapshot('with error mount');
  });
});
