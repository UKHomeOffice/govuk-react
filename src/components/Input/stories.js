import React from 'react';
import { storiesOf } from '@storybook/react';
import Input from '.';

storiesOf('Input', module).add('Component default', () => (
  <Input type="text" />
));
