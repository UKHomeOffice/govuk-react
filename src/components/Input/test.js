import React from "react";
import ReactDOM from "react-dom";
import { mount } from "enzyme";
import Input from "./";

describe("Input", () => {
  const wrapper = <Input type="text" />;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it("matches snapshot", () => {
    expect(mount(wrapper)).toMatchSnapshot(`enzyme.mount`);
  });
});
