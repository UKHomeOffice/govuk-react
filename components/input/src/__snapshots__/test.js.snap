// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches withColouredError snapshot: with coloured error mount 1`] = `
.glamor-0,
[data-glamor-0] {
  box-sizing: border-box;
  font-family: "nta", Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  text-transform: none;
  font-size: 16px;
  line-height: 20px;
  width: 100%;
  padding: 5px 4px 4px;
  border: 4px solid #f47738;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
}

@media only screen and (min-width: 641px) {
  .glamor-0,
  [data-glamor-0] {
    width: 50%;
    font-size: 19px;
    line-height: 25px;
  }
}

<Input
  error="example"
  errorColor="#f47738"
  type="text"
>
  <glamorous(input)
    error="example"
    errorColor="#f47738"
    type="text"
  >
    <input
      className="glamor-0"
      type="text"
    />
  </glamorous(input)>
</Input>
`;

exports[`Input matches withError snapshot: with error mount 1`] = `
.glamor-0,
[data-glamor-0] {
  box-sizing: border-box;
  font-family: "nta", Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  text-transform: none;
  font-size: 16px;
  line-height: 20px;
  width: 100%;
  padding: 5px 4px 4px;
  border: 4px solid #b10e1e;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
}

@media only screen and (min-width: 641px) {
  .glamor-0,
  [data-glamor-0] {
    width: 50%;
    font-size: 19px;
    line-height: 25px;
  }
}

<Input
  error="example"
  type="text"
>
  <glamorous(input)
    error="example"
    type="text"
  >
    <input
      className="glamor-0"
      type="text"
    />
  </glamorous(input)>
</Input>
`;

exports[`Input matches wrapper snapshot: wrapper mount 1`] = `
.glamor-0,
[data-glamor-0] {
  box-sizing: border-box;
  font-family: "nta", Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  text-transform: none;
  font-size: 16px;
  line-height: 20px;
  width: 100%;
  padding: 5px 4px 4px;
  border: 2px solid #0b0c0c;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
}

@media only screen and (min-width: 641px) {
  .glamor-0,
  [data-glamor-0] {
    width: 50%;
    font-size: 19px;
    line-height: 25px;
  }
}

<Input
  type="text"
>
  <glamorous(input)
    type="text"
  >
    <input
      className="glamor-0"
      type="text"
    />
  </glamorous(input)>
</Input>
`;
